### DISCUSSION TITLE
How many game objects is too many?

### DISCUSSION THREAD

**Discussion by tyfai** [replit/kaboom#620] [General]
Hi,

I love the library, it's so refreshingly easy to use. However I have a question:

How many game objects is too many?

It is reasonable to expect this library to be able to handle 100 game objects at once? How about 1000? Would 10000 cause issues?

The reason I'm asking is that I am building a top down shooter with a large map. The map is approximately 48 x 32 tiles in size, which means there would be at least 1536 game objects present. When I add the player, a few dozen enemies, lots of bullets, and some ammunition the game object count could hit 2000.

Is this library a good choice for my game? Has anyone got examples of large games?

Thanks. :)

**Comment by slmjkdbtl** [comment]
Sorry for the late reply. Kaboom is not the most performant game library, v3000 got some really big performance boost, I think generally 2000 on screen objects is ok, if you have a lot of offscreen objects should also use the `offscreen({ hide: true })` component to make sure they don't get rendered at all

**Comment by tyfai** [comment]
No worries, thanks for getting back to me. I will make sure I am using the offscreen hide setting correctly. :)
